output/influxdb/output.go

```diff
@@ -196,11 +218,89 @@ func (o *Output) Stop() error {
 	o.logger.Debug("Stopping...")
 	defer o.logger.Debug("Stopped!")
 	o.periodicFlusher.Stop()
+	o.Client.Close()
+	return nil
+}
+
+// createBucket creates the configured bucket if it doesn't exist
+func (o *Output) createBucket(ctx context.Context) error {
+	_, err := o.Client.BucketsAPI().FindBucketByName(ctx, o.Config.Bucket.String)
+	if err == nil {
+		// the bucket already exists
+		return nil
+	}
+
+	// TODO: It has the bucket name in the error message. Can we do a better check?
+	if err.Error() != fmt.Sprintf("bucket '%s' not found", o.Config.Bucket.String) {
+		return err
+	}
+
+	org, err := o.Client.OrganizationsAPI().FindOrganizationByName(ctx, o.Config.Organization.String)
+	if err != nil {
+		return err
+	}
+
+	var rr []influxdomain.RetentionRule
+	if o.Config.Retention.Valid {
+		rr = []influxdomain.RetentionRule{
+			{
+				// duration in seconds for how long data will be kept in the database
+				EverySeconds: int64(time.Duration(o.Config.Retention.Duration).Round(time.Second).Seconds()),
+				Type:         influxdomain.RetentionRuleTypeExpire,
+			},
+		}
+	}
+	// create a bucket with the default retention policy
+	_, err = o.Client.BucketsAPI().CreateBucketWithName(ctx, org, o.Config.Bucket.String, rr...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createDatabase18 create a database using the v1 API for the old 1.8+ InfluxDB server versions.
+func (o *Output) createDatabase18(ctx context.Context) error {
+	// it uses the Bucket value following the compatibility mode
+	db := o.Config.Bucket.String
```

Comment:
:warning: I'm not sure about it, but I think we don't have other options. To support both database and bucket we have to normalize them to Bucket.

Database: `http://localhost:8086/mydb?precision=1s`
Bucket: `http://localhost:8086/mybucket?retention=7d`